test/sample-code/fileA.cls

```diff
@@ -0,0 +1,28 @@
+public class fileA {
```

Comment:
Since you are having to create files anyway for testing. I almost wonder if it is worth it to quickly just create an engine that looks in files for a string like //Violation(<RuleName>,<Message>})
to more easily create dummy scenarios instead of having to stub everything. Just a thought.